bin/runtime-common/src/messages.rs

```diff
@@ -525,13 +526,20 @@ pub mod target {
 				message_id,
 				message.data.payload.map_err(drop),
 				|dispatch_origin, dispatch_weight| {
-					ThisCurrency::transfer(
-						dispatch_origin,
-						relayer_account,
-						ThisRuntime::WeightToFee::calc(&dispatch_weight),
-						ExistenceRequirement::AllowDeath,
-					)
-					.map_err(drop)
+					let unadjusted_weight_fee = ThisRuntime::WeightToFee::calc(&dispatch_weight);
+					let fee_multiplier = pallet_transaction_payment::Pallet::<ThisRuntime>::next_fee_multiplier();
```

Comment:
Hmm, while this seems innocent and is definitely what we want to do, I started thinking about potential attack vector towards bridge senders.
The relayers can attempt to increase the fee multiplier (by sending remarks for instance) and only then dispatch the message. Do you think this will be an issue in practice? Does the cost of such attack make it feasbile?

Reply:
If I get everything right, then there's absolutely no reasons for relayers to do that. Basically - this code is executed as a part of delivery transaction. The fee of delivery transaction is computed as `BaseTxFee + PerByteFee * TransactionLength + Multiplier * WeightToFee(BaseDeliveryTransactionWeight + MessageDispatchWeight)`. This transfer compensates `Multiplier * WeightToFee(MessageDispatchWeight)` amount that the relayer has just spent (inside the same transaction). So even if relayer will be increasing multiplier, he won't get any profit, because he has just spent the same amount (computed using increased multiplier) on transaction fees.

Reply:
Ugh, yeah, you're totally right. I was somehow thinking that the payment here might be significantly more than what the relayer does actually pay.